pkg/cmd/update-tls-artifacts/generate-owners/tlsmetadatainterfaces/helpers.go

```diff
@@ -21,38 +23,299 @@ func AnnotationValue(whitelistedAnnotations []certgraphapi.AnnotationValue, key
 				continue
 			}
 
-			certKeyPairs[currCert.SecretLocation] = currCert.CertKeyInfo
+			inClusterCertKeyPairs[currCert.SecretLocation] = currCert.CertKeyInfo
 		}
 		for i := range currPKI.InClusterResourceData.CertificateAuthorityBundles {
 			currCert := currPKI.InClusterResourceData.CertificateAuthorityBundles[i]
-			existing, ok := caBundles[currCert.ConfigMapLocation]
+			existing, ok := inClusterCABundles[currCert.ConfigMapLocation]
 			if ok && !reflect.DeepEqual(existing, currCert.CABundleInfo) {
 				errs = append(errs, fmt.Errorf("mismatch of certificate info for --namespace=%v configmap/%v", currCert.ConfigMapLocation.Namespace, currCert.ConfigMapLocation.Name))
 				continue
 			}
 
-			caBundles[currCert.ConfigMapLocation] = currCert.CABundleInfo
+			inClusterCABundles[currCert.ConfigMapLocation] = currCert.CABundleInfo
 		}
+
+		for _, tlsMetadata := range currPKI.OnDiskResourceData.TLSArtifact {
+			found := false
+			isAlsoInCluster := false
+			for _, certKeyPair := range currPKI.CertKeyPairs.Items {
+				for _, loc := range certKeyPair.Spec.OnDiskLocations {
+					if tlsMetadata.OnDiskLocation.Path == loc.Cert.Path || tlsMetadata.OnDiskLocation.Path == loc.Key.Path {
+						found = true
+						isAlsoInCluster = len(certKeyPair.Spec.SecretLocations) > 0
+						break
+					}
+				}
+				if found {
+					break
+				}
+			}
+			if found {
+				if !isAlsoInCluster {
+					info, err := GetCertKeyPairInfoForOnDiskPath(tlsMetadata.OnDiskLocation)
+					if err == nil {
+						onDiskCertsWithoutSecrets[tlsMetadata] = info
+					}
+				}
+				continue
+			}
+
+			for _, caBundle := range currPKI.CertificateAuthorityBundles.Items {
+				for _, loc := range caBundle.Spec.OnDiskLocations {
+					if tlsMetadata.OnDiskLocation.Path == loc.Path {
+						found = true
+						isAlsoInCluster = len(caBundle.Spec.ConfigMapLocations) > 0
+						break
+					}
+				}
+				if found {
+					break
+				}
+			}
+			if found {
+				if !isAlsoInCluster {
+					info, err := GetCertificateAuthorityInfoForOnDiskPath(tlsMetadata.OnDiskLocation)
+					if err == nil {
+						onDiskCABundlesWithoutConfigMaps[tlsMetadata] = info
+					}
+				}
+				continue
+			}
+
+		}
+
 	}
 	if len(errs) > 0 {
 		return nil, utilerrors.NewAggregate(errs)
 	}
 
-	return certs.CertsToRegistryInfo(certKeyPairs, caBundles), nil
+	return certs.CertsToRegistryInfo(inClusterCertKeyPairs, onDiskCertsWithoutSecrets, inClusterCABundles, onDiskCABundlesWithoutConfigMaps), nil
+}
+
+// TODO[vrutkovs]: move this to /api?
+var (
+	onDiskCertificateAuthorities = []certgraphapi.PKIRegistryOnDiskCABundle{
+		{
+			OnDiskLocation: certgraphapi.OnDiskLocationWithMetadata{
+				OnDiskLocation: certgraphapi.OnDiskLocation{
+					Path: "/etc/kubernetes/ca.crt",
+				},
+				User:           "root",
+				Group:          "root",
+				Permissions:    "-rw-r--r--",
+				SELinuxOptions: "system_u:object_r:kubernetes_file_t:s0",
```

Comment:
this information should be coming from the raw tls data.

Reply:
This is the expected file properties, raw TLS data will contain actual data